lib/rules/lighter-video.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @fileoverview Rule to avoid video 
+ * @author Aleksandra Borowska
+ */
+
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Calculate carbon emissions and video duration for embedded YouTube videos',
+      recommended: true,
+      url: "https://ecolint.org/docs/rules/lighter-video"
+    },
+    schema: [
+
+    ]
+  },
+  create: function (context) {
+
+    const emissionsPerMinute = {
+      small: 0.8,
+      medium: 1.0,
+      large: 1.2,
+      hd720: 1.5,
+      hd1080: 2.0,
+      highres: 2.5,
+    };
+
+    function getPropertyValue(obj, property) {
+      const prop = obj.properties.find(prop => prop.key.name === property);
+      return prop ? prop.value.value : null;
+    }
+
+    function calculateCarbonEmissions(quality, playbackTime) {
+    
+      const emissionsPerSecond = emissionsPerMinute[quality] / 60;
+      const carbonEmissions = emissionsPerSecond * playbackTime;
+
+      return { carbonEmissions};
+    }
+
+    return {
+      "CallExpression[callee.type='MemberExpression'][callee.property.name=/cueVideoById|loadVideoById|loadVideoByUrl/][arguments.length>0]": function (node) {
+        const methods = ["cueVideoById", "loadVideoById", "loadVideoByUrl"];
+        const { callee, arguments: args } = node;
+        if (
+          (callee.type === "MemberExpression" &&
```

Comment:
let's export the inside of if to separate two functions with proper names, so it will be easily readable

Reply:
ok, done.

Reply:
also no changes here, I will mark as unresolved